src/merger/merger.cpp

```diff
@@ -0,0 +1,968 @@
+            classes_map,
+            turn_lane_map,
+            scripting_environment,
+            it->first,
+            number_of_threads,
+            class_names,
+            excludeable_classes_set);
+        it++;
+    }
+
+    std::vector<std::vector<std::string>> excludeable_classes;
+    for (const auto &combination_set : excludeable_classes_set)
+    {
+        excludeable_classes.push_back(std::vector<std::string>(combination_set.begin(), combination_set.end()));
+    }
+
+    writeTimestamp();
+    // use first scripting_environment as profile accumulator
+    writeOSMData(
+        extraction_containers,
+        classes_map,
+        class_names,
+        excludeable_classes,
+        scripting_environment_first);
+
+    TIMER_STOP(extracting);
+    util::Log() << "extraction finished after " << TIMER_SEC(extracting) << "s";
+
+    std::vector<extractor::TurnRestriction> turn_restrictions = std::move(extraction_containers.turn_restrictions);
+    std::vector<extractor::UnresolvedManeuverOverride> unresolved_maneuver_overrides = std::move(extraction_containers.internal_maneuver_overrides);
+
+    // Transform the node-based graph that OSM is based on into an edge-based graph
+    // that is better for routing.  Every edge becomes a node, and every valid
+    // movement (e.g. turn from A->B, and B->A) becomes an edge
+    util::Log() << "Generating edge-expanded graph representation";
+
+    TIMER_START(expansion);
+
+    extractor::EdgeBasedNodeDataContainer edge_based_nodes_container;
+    std::vector<extractor::EdgeBasedNodeSegment> edge_based_node_segments;
+    util::DeallocatingVector<extractor::EdgeBasedEdge> edge_based_edge_list;
+    std::vector<EdgeWeight> edge_based_node_weights;
+    std::vector<EdgeDuration> edge_based_node_durations;
+    std::vector<EdgeDistance> edge_based_node_distances;
+    std::uint32_t ebg_connectivity_checksum = 0;
+
+    /*
+        TODO: Use first scripting environment as only .processTurn(...) and .GetProfileProperties().GetWeightMultiplier()
+        are used
+    */
+    // Create a node-based graph from the OSRM file
+    extractor::NodeBasedGraphFactory node_based_graph_factory(
+        config.GetPath(".osrm"),
+        scripting_environment_first,
+        turn_restrictions,
+        unresolved_maneuver_overrides);
+
+    extractor::NameTable name_table;
+    extractor::files::readNames(config.GetPath(".osrm.names"), name_table);
+
+    util::Log() << "Find segregated edges in node-based graph ..." << std::flush;
+    TIMER_START(segregated);
+
+    auto segregated_edges = guidance::findSegregatedNodes(node_based_graph_factory, name_table);
+
+    TIMER_STOP(segregated);
+    util::Log() << "ok, after " << TIMER_SEC(segregated) << "s";
+    util::Log() << "Segregated edges count = " << segregated_edges.size();
+
+    util::Log() << "Writing nodes for nodes-based and edges-based graphs ...";
+    auto const &coordinates = node_based_graph_factory.GetCoordinates();
```

Comment:
This can be changed in order to be consistent with the rest to `const auto`.

Reply:
Done.